nilearn/maskers/surface_maps_masker.py

```diff
@@ -0,0 +1,426 @@
+                )
+            else:
+                concat_data[key] = None
+
+        return concat_data["maps"], concat_data["mask"]
+
+    def transform(self, img, confounds=None, sample_mask=None):
+        """Extract signals from surface object.
+
+        Parameters
+        ----------
+        img : :obj:`~nilearn.surface.SurfaceImage` object or \
+              :obj:`list` of :obj:`~nilearn.surface.SurfaceImage` or \
+              :obj:`tuple` of :obj:`~nilearn.surface.SurfaceImage`
+            Mesh and data for both hemispheres/parts. The data for each \
+            hemisphere is of shape (n_vertices/2, n_timepoints).
+
+        confounds : :class:`numpy.ndarray`, :obj:`str`,\
+                    :class:`pathlib.Path`, \
+                    :class:`pandas.DataFrame` \
+                    or :obj:`list` of confounds timeseries, default=None
+            Confounds to pass to :func:`nilearn.signal.clean`.
+
+        sample_mask : None, Any type compatible with numpy-array indexing, \
+                  or :obj:`list` of \
+                  shape: (number of scans - number of volumes removed) \
+                  for explicit index, or (number of scans) for binary mask, \
+                  default=None
+            sample_mask to pass to :func:`nilearn.signal.clean`.
+
+
+        Returns
+        -------
+        region_signals: :obj:`numpy.ndarray`
+            Signal for each region as provided in the maps (via `maps_img`).
+            shape: (n_timepoints, n_regions)
+        """
+        self._check_fitted()
+
+        # if img is a single image, convert it to a list
+        # to be able to concatenate it
+        if not isinstance(img, list):
+            img = [img]
+        img = concatenate_surface_images(img)
+        # check img data is 2D
+        check_surface_data_ndims(img, 2, "img")
+        check_same_n_vertices(self.maps_img.mesh, img.mesh)
+        img_data = np.concatenate(
+            list(img.data.parts.values()), axis=0
+        ).astype(np.float32)
+
+        # get concatenated hemispheres/parts data from maps_img and mask_img
+        maps_data, mask_data = self._get_concatenated_maps_and_mask
+
+        if self.smoothing_fwhm is not None:
+            warnings.warn(
+                "Parameter smoothing_fwhm "
+                "is not yet supported for surface data",
+                UserWarning,
+                stacklevel=2,
+            )
+            self.smoothing_fwhm = None
+
+        parameters = get_params(
+            self.__class__,
+            self,
+        )
+        if self.clean_args is None:
+            self.clean_args = {}
+        parameters["clean_args"] = self.clean_args
+
+        if self.memory is None:
+            self.memory = Memory(location=None)
+
+        # apply mask if provided
+        # and then extract signal via least square regression
+        if mask_data is not None:
+            region_signals = cache(
+                linalg.lstsq,
```

Comment:
Just wondering what happens if the least squares fit fails, due to redundancy in the Surface maps ?

Reply:
I think scipy handle that and would throw a LinAlgError:

https://github.com/scipy/scipy/blob/92d2a8592782ee19a1161d0bf3fc2241ba78bb63/scipy/linalg/_basic.py#L1191